### ISSUE TITLE
Failed to find Minecraft main class

### ISSUE THREAD

**Issue by TheBookAjah** [GTNewHorizons/GT-New-Horizons-Modpack#12834]
### Your GTNH Discord Username

The Book Ajah#5563

### Your Pack Version

2.3.0-RC-3

### Your Server

SP

### Java Version

Java 17

### Type of Server

None

### Your Actions

Tried to start the game and it crashed. I've tried multiple ways to update my Java 8 GTNH instance to Java 9+, including updating it normally and creating a new instance with just my world and other bits that the wiki advised to port over kept, but every time it has the same crash. The Java 8 instance still works perfectly. I'm using Java 17.0.6. As far as I know, I've followed the right procedure for updating to Java 9+ with all the instructions. This happened with RC-2 as well as RC-3. (So far I've ended up going back to Java 8 for the time being.)
Basically crash on startup.

### Crash Report

https://paste.ee/p/H6Czo

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this crash consistently by following the exact steps I described above.
- [ ] I have asked other people and they confirm they also crash by following the exact steps I described above.

**Comment by Glease** [comment]
Check the `C:/Users/P320/Documents/MultiMC/mmc-stable-win32/MultiMC/libraries/net/minecraft/launchwrapper/1.15/launchwrapper-1.15.jar` file on your disk. It should have a size of 30875 byte and can be opened as a zip file(*). You most likely got a broken jar here. Delete it to let MultiMC download one for you, or get it [here](http://jenkins.usrv.eu:8081/nexus/content/groups/public/net/minecraft/launchwrapper/1.15/launchwrapper-1.15.jar) yourself and put it at that location.

(*) if you make a copy and rename the copy to launchwrapper-1.15.zip, you can open it with double click and your file manager will not complain about broken archive.

**Comment by TheBookAjah** [comment]
It does seem to be working now, but interestingly only after I used the manually downloaded file. Strangely, the file downloaded by MultiMC didn't work, and I couldn't actually open it as an archive with 7-zip or with the method you put above. The one from the link you provided works nicely though, so thank you!

**Comment by Glease** [comment]
Probably multimc download went corrupted somehow. IDK. It's as if only IDM and browsers know how to not download a corrupted files these day. In any case, case closed as it's a mere download corruption.
